### PR TITLE
Fix mean in `AffineTransformedDistribution`

### PR DESCRIPTION
**Pull request by stailx**
Incorrect mean property in AffineTransformedDistribution.
Scaling was not taken into account in an Affine Transform.
Mathematic proof: mean(y)= mean(scale * x+loc ) = scale * mean(x) + loc.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/mx/distribution/transformed_distribution.py [hunk 1]
```diff
@@ -210,9 +210,9 @@ def __init__(
     @property
     def mean(self) -> Tensor:
         return (
-            self.base_distribution.mean
+            self.base_distribution.mean * self.scale
```
Comment (lostella) [inline src/gluonts/mx/distribution/transformed_distribution.py:213]: This does not account for the case `self.scale = None`. You could use `scale` defined as

```python
scale = self.scale if self.scale is not None else 1
```

**Review by lostella — CHANGES_REQUESTED** [review]
@stailx thanks for spotting this! I only have one comment to be addressed

**Review by lostella — APPROVED** [review]
@stailx thanks again for spotting this! I took the liberty of pushing this over the finish line, I hope you don't mind

🚀

**Comment by kashif** [comment]
well spotted @stailx

**Pull request merged**